unified-runtime/source/adapters/level_zero/v2/queue_immediate_in_order.cpp

```diff
@@ -767,18 +769,19 @@ ur_result_t ur_queue_immediate_in_order_t::bindlessImagesImageCopyExp(
     ur_exp_image_copy_region_t *pCopyRegion,
     ur_exp_image_copy_flags_t imageCopyFlags, uint32_t numEventsInWaitList,
     const ur_event_handle_t *phEventWaitList, ur_event_handle_t *phEvent) {
-  std::ignore = pDst;
-  std::ignore = pSrc;
-  std::ignore = pSrcImageDesc;
-  std::ignore = pDstImageDesc;
-  std::ignore = imageCopyFlags;
-  std::ignore = pSrcImageFormat;
-  std::ignore = pDstImageFormat;
-  std::ignore = pCopyRegion;
-  std::ignore = numEventsInWaitList;
-  std::ignore = phEventWaitList;
-  std::ignore = phEvent;
-  return UR_RESULT_ERROR_UNSUPPORTED_FEATURE;
+
+  auto commandListMgr = commandListManager.lock();
+
+  auto zeSignalEvent =
+      getSignalEvent(commandListMgr, phEvent, UR_COMMAND_MEM_IMAGE_COPY);
+  auto waitListView =
+      getWaitListView(commandListMgr, phEventWaitList, numEventsInWaitList);
+
+  return bindlessImagesHandleCopyFlags(
```

Comment:
A bit out of the scope of this review but out of interest here do you know how much overhead is added to `bindlessImagesHandleCopyFlags` if the optional event parameters are provided/vs not provided: e.g.  `zeSignalEvent`/`waitListView`?

I am investigating this for performance optimization of the `nd_launch` API which does not provide/return events.

Reply:
We recently added eventless enqueue benchmarks, see results here:
https://github.com/intel/llvm/pull/17898#issuecomment-2785559156

There's a noticeable difference, but it's not major.